sdk/Managed/test/Microsoft.WindowsAzure.MobileServices.Test/UnitTests/Table/Sync/MobileServiceSyncContext.Test.cs

```diff
@@ -246,6 +246,136 @@ public class MobileServiceSyncContextTests : TestBase
             await service.SyncContext.PushAsync();
         }
 
+        class LocalStoreWithDelay : MobileServiceLocalStoreMock
+        {
+            private int _lookupDelayInMilliseconds = 0;
+
+            public override async Task<JObject> LookupAsync(string tableName, string id)
+            {
+                if (0 != _lookupDelayInMilliseconds)
+                {
+                    // releases this thread and causes this lookup to take longer
+                    await Task.Delay(_lookupDelayInMilliseconds);
+                }
+
+                return await base.LookupAsync(tableName, id);
+            }
+
+            /// <summary>
+            /// Tells this mock to yield and to delay for the given number of milliseconds before proceeding with the LookupAsync().
+            /// </summary>
+            /// <param name="delayInMilliseconds">The delay in milliseconds.</param>
+            internal void SetLookupDelay(int delayInMilliseconds)
+            {
+                _lookupDelayInMilliseconds = delayInMilliseconds;
+            }
+        }
+
+        [AsyncTestMethod]
+        public async Task PushAsync_Succeeds_WithPendingOperations_AndOpQueueIsConsistent()
```

Comment:
Blocking on DeleteAsync() in the store gives OperationQueue.DeleteAsync() a chance to call GetOperationAsync() first. To ensure the conditions necessary to prove what this test proves, it's necessary to inject the delay as early as possible in OperationQueue.DeleteAsync(), and the first thing that happens, effectively, is the LookupAsync() call on the local store.